app.py

```diff
@@ -73,7 +74,10 @@ def signup():
             phone_number = request.form.get("phone_number")
             password = request.form.get("password")
 
-            hashed_password = hash_password(password)
+            if not password:  # Check if password is provided
+                return render_template("signup.html", message="Password is required")
+
+            hashed_password = generate_password_hash(password)
```

Comment:
yea, u guys changed the logic of hashed password, as long as u guys able to understand how the code works, im cool with all changes